src/planner/ompl/Planner.cpp

```diff
@@ -372,6 +374,100 @@ trajectory::InterpolatedPtr planCRRTConnect(
       std::move(_interpolator),
       _maxPlanTime);
 }
+
+//=============================================================================
+std::pair <std::unique_ptr<trajectory::Interpolated>, bool> simplifyOMPL(statespace::StateSpacePtr _stateSpace,
+                                                                    statespace::InterpolatorPtr _interpolator,
+                                                                    distance::DistanceMetricPtr _dmetric,
+                                                                    constraint::SampleablePtr _sampler,
+                                                                    constraint::TestablePtr _validityConstraint,
+                                                                    constraint::TestablePtr _boundsConstraint,
+                                                                    constraint::ProjectablePtr _boundsProjector, 
+                                                                    double _maxDistanceBtwValidityChecks,
+                                                                    double _timeout, size_t _maxEmptySteps,
+                                                                    trajectory::InterpolatedPtr _originalTraj)
+{
+
+  // Step 1: Generate the space information of OMPL type
+  auto si = getSpaceInformation(
+        _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+        std::move(_validityConstraint), std::move(_boundsConstraint),
+        std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+  
+  // Step 2: Convert AIKIDO Interpolated Trajectory to OMPL path 
+  auto path = toOMPLTrajectory(_originalTraj, si);
+
+  // Step 3: Use the OMPL methods to simplify the path
+  ::ompl::geometric::PathSimplifier simplifier {si};
+
+  // Flag for user to know if shorten was successful
+  bool shorten_success = false;
+
+  // Set the parameters for termination of simplification process
+  std::chrono::system_clock::time_point const time_before = std::chrono::system_clock::now(); 
+  std::chrono::system_clock::time_point time_current;
+  std::chrono::duration<double> const time_limit = std::chrono::duration<double>(_timeout); 
+  double empty_steps = 0;
+  
+  do 
+  {
+
+    bool const shortened = simplifier.shortcutPath(path, 1, _maxEmptySteps, 1.0, 0.0);
+    empty_steps = (empty_steps + !shortened)*!shortened; // Increment only if shorten fails consecutively
+    time_current = ::ompl::time::now();
```

Comment:
This line causes a [compilation error](https://travis-ci.org/personalrobotics/aikido/builds/223049697#L2624). It should be replaced to `time_current = std::chrono::system_clock::now();`.